add controllers

CrazyTapok opened the pull request:
1. Screenshot: 
![image](https://github.com/user-attachments/assets/ccb6a331-60fa-4c25-9fb4-32b3dba693fb)


2. What has been implemented: 
  - [x] add controllers
  - [x] add request/response models
  - [x] add test for controllers
  - [x] change access modifier for IEmployeeService, IProjectService, ITaskService

CrazyTapok committed:
feat: add controllers
add controllers, add request/response models, add test for controllers, change access modifier for IEmployeeService, IProjectService, ITaskService

svirzhevskiy commented on src/TaskManager.API/Contracts/Request/CompanyRequest.cs:
```diff
@@ -0,0 +1,6 @@
+﻿namespace TaskManager.API.Contracts.Request;
+
+public class CompanyRequest
```
можно воспользоваться record для таких моделей. Почитай про них, кароч)

svirzhevskiy commented on src/TaskManager.API/Contracts/Request/EmployeeRequest.cs:
```diff
@@ -0,0 +1,12 @@
+﻿using TaskManager.Core.Enums;
+
+namespace TaskManager.API.Contracts.Request;
```
мб тогда папки назвать во множественном числе?

svirzhevskiy commented on src/TaskManager.API/Controllers/CompanyController.cs:
```diff
@@ -0,0 +1,96 @@
+namespace TaskManager.API.Controllers;
+
+[ApiController]
+[Route("[controller]")]
```
не хош url "api/[controller]"? А так мне урлы нравятся, по ресту)

CrazyTapok replied:
а ведь изначально он был  "api/[controller]")

svirzhevskiy commented on src/TaskManager.API/Controllers/CompanyController.cs:
```diff
@@ -0,0 +1,96 @@
+            return NotFound();
+        }
+
+        var response = new CompanyResponse
```
Не хочешь как-то вынести маппинг из контроллеров и сервисов? Мб в отдельный сервис, в конструктор, в оператор каста или в экстеншн метод

CrazyTapok replied:
это сделать для моделей response/request или еще для моделей Entity?

svirzhevskiy replied:
для всего

svirzhevskiy commented on src/TaskManager.API/Controllers/CompanyController.cs:
```diff
@@ -0,0 +1,96 @@
+
+        var result = await _companyService.UpdateAsync(company, cancellationToken);
+       
+        if (!result)
```
не нравится тут название result. Читаешь как "not result" - не понятно мне

CrazyTapok replied:
Согласен, исправлю

svirzhevskiy commented on src/TaskManager.API/Controllers/EmployeeController.cs:
```diff
@@ -0,0 +1,143 @@
+        return NoContent();
+    }
+
+    [HttpGet("project/{projectId:guid}")]
```
тут урл не нравится, не по ресту получается. Мб что-то такое хочется `projects/{projectId}/employees`, подумай, кароч)

svirzhevskiy replied:
надо почитать, поискать кейсы как делают в таких случаях, мб у гпт спросить)

CrazyTapok committed:
refactor: fix remarks after the review
using record for Response/Request Models, replace the URL "[controller]" with "api/[controller]", mapping move from controllers to extension methods, change the URLs of methods according to REST, replace the name of the "result" parameter to "updateSuccessful", using primary constructors in controllers.

svirzhevskiy commented on src/TaskManager.API/Controllers/EmployeeController.cs:
```diff
@@ -0,0 +1,87 @@
+        return NoContent();
+    }
+
+    [HttpGet("projects/{projectId:guid}/employees")]
+    public async Task<ActionResult<List<EmployeeResponse>>> GetEmployeesByProjectIdAsync(Guid projectId, CancellationToken cancellationToken = default)
+    {
+        var employees = await _employeeService.GetEmployeesByProjectIdAsync(projectId, cancellationToken);
+        var response = employees.Select(employee => employee.MapToEmployeeResponse()).ToList();
+
+        return Ok(response);
+    }
```
я бы в проджект контроллер вынес

svirzhevskiy commented on src/TaskManager.API/Controllers/EmployeeController.cs:
```diff
@@ -0,0 +1,87 @@
+    }
+
+    [HttpGet("companies/{companyId:guid}/employees")]
+    public async Task<ActionResult<List<EmployeeResponse>>> GetEmployeesByCompanyIdAsync(Guid companyId, CancellationToken cancellationToken = default)
+    {
+        var employees = await _employeeService.GetEmployeesByCompanyIdAsync(companyId, cancellationToken);
+        var response = employees.Select(employee => employee.MapToEmployeeResponse()).ToList();
+
+        return Ok(response);
+    }
```
а это в компанис

CrazyTapok committed:
refactor: fix remarks after the review
methods are spaced across controllers according to Rest, corrected tests.
